### PR TITLE
[SwiftUI] Rework SwiftUIMeasurementContainer to use sizeThatFits

### PR DESCRIPTION
**Pull request by erichoracek**
## Change summary
This is the "correct" method to implement the behavior that we were currently implement on iOS 13-15 using a `GeometryReader`. This is evidenced by the fact that Apple made this API public on iOS 16. As such, we move to it instead, and migrate off of the `GeometryReader` approach. Thankfully we're able to keep pretty much the same public facing API.

## How was it tested?
*How did you verify that this change accomplished what you expected? Add more detail as needed.*
- [ ] Wrote automated tests
- [x] Built and ran on the iOS simulator
- [ ] Built and ran on a device

## Pull request checklist
*All items in this checklist must be completed before a pull request will be reviewed.*
- [x] Risky changes have been put behind a feature flag, e.g. `CollectionViewConfiguration`
- [x] Added a [`CHANGELOG.md` entry](https://keepachangelog.com/en/1.0.0/) in the "Unreleased" section for any library changes

### CODE REVIEW

File: Sources/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringUIViewRepresentable.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    var copy = self
+    copy.sizing = strategy
+    return copy
+  }
+}
+
+// MARK: Defaults
+
+extension MeasuringUIViewRepresentable {
+  public func _overrideSizeThatFits(
```
Comment (thedrick) [inline Sources/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringUIViewRepresentable.swift:45]: is this an override function, or it is just expected that SwiftUI / UIKit will call this method?

**Review by thedrick — APPROVED** [review]
LGTM @erichoracek

File: Sources/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringUIViewRepresentable.swift (hunk 1 above)
Comment (erichoracek) [reply Sources/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringUIViewRepresentable.swift:45]: SwiftUI will just call this for us 👍 — it just has a weird name

File: Sources/EpoxyCore/SwiftUI/LayoutUtilities/SwiftUIMeasurementContainer.swift [hunk 2]
```diff
@@ -35,55 +43,75 @@ public final class SwiftUIMeasurementContainer<SwiftUIView, UIViewType: UIView>:
...
 
+  /// The  most recently measured fitting size of the `uiView` that fits within the current
+  /// `proposedSize`, else `zero` if it has not yet been measured.
+  public private(set) var measuredFittingSize = CGSize.zero
+
+  /// The most recently measured intrinsic content size of the `uiView`, else `noIntrinsicMetric` if
+  /// it has not yet been measured.
+  ///
+  /// Contains `UIView.noIntrinsicMetric` fields for dimensions with no intrinsic size.
+  public private(set) var measuredIntrinsicContentSize = CGSize.noIntrinsicMetric
```
[the remark this answers is not in the document]
Comment (erichoracek) [reply Sources/EpoxyCore/SwiftUI/LayoutUtilities/SwiftUIMeasurementContainer.swift]: Good point, I renamed this `_intrinsicContentSize` and added a clarification doc comment

**Pull request merged**